Astro CLI 1.3.0 core doc changes (#455)
astronomer/docs: astro/cli-reference/astrocloud-dev-parse.md

---
sidebar_label: "astrocloud dev parse"
title: "astrocloud dev parse"
id: astrocloud-dev-parse
description: Reference documentation for astrocloud dev parse.
---

## Description

> Parse the DAGs in a locally hosted Astro project. For more information about testing DAGs locally, read [Run Tests with the Astro CLI](test-and-troubleshoot-locally.md#run-tests-with-the-astro-cli).

Reviewer: ```suggestion
Parse the DAGs in a locally hosted Astro project to quickly check them for errors. For more information about testing DAGs locally, read [Run Tests with the Astro CLI](test-and-troubleshoot-locally.md#run-tests-with-the-astro-cli).
```

---
sidebar_label: "astrocloud dev parse"
title: "astrocloud dev parse"
id: astrocloud-dev-parse
description: Reference documentation for astrocloud dev parse.
---

## Description

> Run several tests on your project code. These tests call all declared OS-level variables, Airflow Connections, and Airflow Variables to check whether each of those objects has a value.
> 
> Additionally, this command runs prebuilt unit tests on your Astro project using Pytest. These unit tests are the same ones that are included by default with your project as described in [Run Tests with Pytest](test-and-troubleshoot-locally.md#run-tests-with-pytest).

Reviewer 2: ```suggestion
Run basic tests against your Astro project in a local Airflow environment. Specifically, this command runs two tests:

 - A test that parses all references to environment variables, Airflow Connections, and Airflow Variables in your DAGs to verify that each of those objects has a value. This ensures that your DAGs don't call an object that does not exist.
 - The DAG integrity test that's included in your Astro project by default and is run on `astrocloud dev pytest`. 
 
For more information on testing DAGs locally, read [Run Tests with the Astro CLI](test-and-troubleshoot-locally.md#run-tests-with-the-astro-cli).
```

Reviewer 2: @jwitz TBH I almost feel like you could remove this detail and just link out to the CLI doc at the end (the H2 header that covers both pytest and parse)

Reviewer: Honestly I think we should just say that this command just lets you know if your DAGs have errors or not.(syntax import, processable DAG code etc...). We can say that it is running a test located `.astrocloud/test_dag_integrity_default.py` but don't need to say exactly how it works. Also it explicitly does not give errors for missing variables and connections. Many users said that this one thing they may not have locally but they don't want it stop their deploys

## Usage

```sh
astrocloud dev parse
```

## Options

| Option              | Description                                                                                   | Possible Values                                 |
| ------------------- | --------------------------------------------------------------------------------------------- | ----------------------------------------------- |

> | `-e`, `--env`       | The filepath to your environment variables. The default is `.env`)                            | Any valid filepath within your Astro project    |

Reviewer 2: ```suggestion
| `-e`, `--env`       | The filepath to your environment variables. The default is `.env`.                            | Any valid filepath within your Astro project    |
```
